include/seqan/index/index_bidirectional.h

```diff
@@ -0,0 +1,187 @@
+struct RevTextFibre<ModifiedString<TText, ModReverse> >
+{
+    typedef TText Type;
+};
+
+template <typename TText, typename TTextConfig>
+struct RevTextFibre<StringSet<TText, TTextConfig> >
+{
+    typedef StringSet<typename RevTextFibre<TText>::Type, TTextConfig> Type;
+};
+
+// ----------------------------------------------------------------------------
+// Class BidirectionalIndex
+// ----------------------------------------------------------------------------
+
+/*!
+ * @class BidirectionalIndex
+ * @headerfile <seqan/index.h>
+ * @brief A bidirectional index class.
+ *
+ * @signature template <typename TText, typename TIndexSpec>
+ *            class Index<TText, Bidirectional<TIndexSpec> >;
+ *
+ * @tparam TText           The text type. Types: @link String @endlink, @link StringSet @endlink
+ * @tparam TIndexSpec   Unidirectional index used for constructing a bidirectional one.
+ *
+ * @section Structure
+ *
+ * The BidirectionalIndex consists of two unidirectional indices that index the original text and the reverse text.
+ */
+
+template <typename TText, typename TIndexSpec>
+class Index<TText, BidirectionalIndex<TIndexSpec> >
+{
+    typedef typename RevTextFibre<TText>::Type	TRevText;
+    typedef Index<TRevText, TIndexSpec>         TRevIndex;
+    typedef Index<TText, TIndexSpec>            TFwdIndex;
+
+    public:
+
+    TRevText	revText;
+    TRevIndex   rev;
+    TFwdIndex   fwd;
+
+    Index()    {}
+
+    Index(TText & text) :
+        revText(text),
+        rev(revText),
+        fwd(text)
+    {}
+};
+
+// ============================================================================
+// Functions
+// ============================================================================
+
+// ----------------------------------------------------------------------------
+// Function clear()
+// ----------------------------------------------------------------------------
+
+// Already documented
+template <typename TText, typename TIndexSpec>
+inline void clear(Index<TText, BidirectionalIndex<TIndexSpec> > & index)
+{
+    clear(index.fwd);
+    clear(index.rev);
+    clear(index.revText);
+}
+
+// ----------------------------------------------------------------------------
+// Function empty()
+// ----------------------------------------------------------------------------
+
+// This function checks whether the index is empty. Its already documented.
+template <typename TText, typename TIndexSpec>
+inline bool empty(Index<TText, BidirectionalIndex<TIndexSpec> > const & index)
+{
+    return empty(index.fwd) || empty(index.bwd);
+}
+
+// This function can be used to open a previously saved index.
+template <typename TText, typename TIndexSpec>
+inline bool open(Index<TText, BidirectionalIndex<TIndexSpec> > & index, const char * fileName)
+{
+    String<char> name;
+
+    name = fileName;    append(name, ".fwd");
+    bool fwdIndex = open(index.fwd, toCString(name), DefaultOpenMode<Index<TText, TIndexSpec> >::VALUE);
+    if (fwdIndex)
```

Comment:
It's more readable as in https://github.com/seqan/seqan/blob/master/include/seqan/index/index_fm.h#L466 :nail_care: